### PR TITLE
Add approveForAllWithSignature functionality to xcert.

### PR DESCRIPTION
**Pull request by MoMannn**
TODO: 

- [x] Charge erc20 token for not eth approve

### CODE REVIEW

File: packages/0xcert-ethereum-xcert-contracts/src/contracts/xcert.sol [hunk 1]
```diff
@@ -199,6 +244,134 @@ contract XcertToken is
...
+    );
+    require(!claimPerformed[claim], CLAIM_PERFORMED);
+    require(_expiration >= now, CLAIM_EXPIRED);
+    claimPerformed[claim] = true;
+    ownerToOperators[_owner][_operator] = _approved;
+    emit ApprovalForAll(_owner, _operator, _approved);
+  }
+
+  /**
+   * Generates hash of the set approval for.
```
Comment (xpepermint) [inline packages/0xcert-ethereum-xcert-contracts/src/contracts/xcert.sol]: `@dev`

File: packages/0xcert-ethereum-xcert-contracts/src/contracts/mocks/xcert-mock.sol [hunk 2]
```diff
@@ -37,7 +38,7 @@ contract XcertMock is XcertToken {
     {
       supportedInterfaces[_capabilities[i]] = true;
     }
-    addressToAbility[msg.sender] = 2047; // Assigns all available abilities to creator. 
+    addressToAbility[msg.sender] = 2047; // Assigns all available abilities to creator.
```
Comment (fulldecent) [inline packages/0xcert-ethereum-xcert-contracts/src/contracts/mocks/xcert-mock.sol]: ? `~0x0`

Comment (MoMannn) [reply packages/0xcert-ethereum-xcert-contracts/src/contracts/mocks/xcert-mock.sol]: I just removed an unnecessary backspace.

**Comment by MoMannn** [comment]
Two tests are skipped because: https://github.com/trufflesuite/truffle/issues/2449

**Pull request merged**